File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>


int main(int argc, string argv[])
{
  if (argc != 2)
  {
    printf("Error: caesar should be called with 1 command line argument\n");
    return 1;
  }
  else
  {
    //convert key to int and wrap it around
    int k = atoi(argv[1]) % 26;
    //get plaintext from user
    printf("plaintext: ");
    string p = get_string();
    
    printf("ciphertext: ");

    for (int i = 0, n = strlen(p); i < n; i++){
      //encipher
      if (isupper(p[i]))
      {
        printf("%c",(((p[i] - 'A') + k) % 26) + 'A');
      }
      else if (islower(p[i]))
      {
        printf("%c",(((p[i] - 'a') + k) % 26) + 'a');
      }

      // deal with non alphabetical symbols
      else
      {
        printf("%c",p[i]);
      }
    }
    printf("\n");
    return 0;
  }
}

File: pset6/caesar.py
# encrypt messages using caesar's cipher
# http://docs.cs50.net/problems/caesar/caesar.html

import sys

def main():
    # check correct usage
    if len(sys.argv) != 2:
        sys.exit('Usage: caesar.py 13');

    # convert key to int and wrap it around alphabet
    k = int(sys.argv[1]) % 26

    # get plaintext from user
    p = input('plaintext: ')

    # record ASCI positions of A & a
    intA = ord('A')
    inta = ord('a')

    # set up output for enicphered text
    print('ciphertext: ', end = '');

    for i in range(len(p)):
        # if alphabetical character, encrypt. Otherwise leave alone
        if p[i].isupper():
            print(chr((((ord(p[i]) - intA) + k) % 26) + intA), end = '')
        elif p[i].islower():
            print(chr((((ord(p[i]) - inta) + k) % 26) + inta), end = '')
        else:
            print(p[i], end = '')
    print('');
if __name__ == "__main__":
    main()
